testing/test_remote.py

```diff
@@ -79,6 +79,21 @@ def test_remoteinitconfig(testdir):
 
 
 class TestReportSerialization:
+    def test_xdist_longrepr_to_str_issue_241(self, testdir):
```

Comment:
I would have preferred doing a test which just test that the report was not mutated before and after both serialization and serialization. I couldn't find a elegant python way of doing this so I am open to suggestions.

Reply:
It is hard to properly test that because unfortunately that involves a lot of internal pytest structures which are hard to create. I think this test is enough, I verified it reproduces the issue nicely.